packages/mdc-menu/simple/foundation.js

```diff
@@ -91,8 +91,10 @@ export default class MDCSimpleMenuFoundation extends MDCFoundation {
     this.keyupHandler_ = (evt) => this.handleKeyboardUp_(evt);
     /** @private {function(!Event)} */
     this.documentClickHandler_ = (evt) => {
-      this.adapter_.notifyCancel();
-      this.close(evt);
+      if (!evt.target.classList.contains('mdc-list-item')) {
```

Comment:
I typically walk all the way up the tree until I get to the `body` or `documentElement`. It shouldn't really impact performance at all, and it ensures correct behavior in all cases.